src/main/kotlin/com/personio/synthetics/builder/api/StepBuilder.kt

```diff
@@ -32,30 +34,45 @@ class StepBuilder(
 
     private val step = SyntheticsAPIStep(SyntheticsAPITestStep())
     private var request: SyntheticsTestRequest? = null
+    private var waitDuration: Int? = null
     private var assertions = listOf<SyntheticsAssertion>()
     private val parsingOptions = mutableListOf<SyntheticsParsingOptions>()
 
     fun build(): SyntheticsAPIStep {
-        if (request == null) {
-            throw IllegalStateException("Request must be provided.")
+        if (request == null && waitDuration == null) {
+            throw IllegalStateException("Provide either of Request or Wait duration.")
+        }
+        if (request != null && waitDuration != null) {
+            throw IllegalStateException("Only one of Request or Wait duration should be provided.")
         }
-
         if (allowFailure) {
             step.syntheticsAPITestStep.isCritical(isCritical)
         }
 
         if (parsingOptions.isNotEmpty()) {
             step.syntheticsAPITestStep.extractedValues(parsingOptions)
         }
+        return if (waitDuration != null) {
+            buildWaitStep()
+        } else {
+            SyntheticsAPIStep(
```

Comment:
nit: Can we also pull this in a function? Then we can simplify this if else with a ternary operator for example.